### PR TITLE
Shushman/1.0 fixes

### PR DESCRIPTION
**Pull request by Shushman**
Several bug fixes about file saving in `tmp` dirs that were breaking `PGFPlots`. Essentially, with the multiple calls to `mktempdir`,  nesting of temp dirs was happening, and the `tmp_10000001.png` files generated by `PGFPlots` were not being found.

I have basically changed it to output everything to the current directory or the directory preceding the filename, and then switch back to the current directory after being done. Please review

### CODE REVIEW

**Comment by mykelk** [comment]
This looks good to me. @tawheeler it looks like you added the `mktempdir` calls? It seems to break things by creating a bunch of nested temp directories. If you agree with these changes, you can merge.

File: src/TikzPictures.jl [hunk 1]
```diff
@@ -217,13 +216,9 @@ function save(f::PDF, tp::TikzPicture)
...
         # This failing is NOT critical either, so just make it a warning
         if tikzDeleteIntermediate()
-            target_file = temp_dir * "/" * "$(f.filename).pdf"
-            if isfile(target_file)
-                # Note: file may not exist if !latexSuccess
-                # Moves pdf out of temp directory
-                mv(target_file, foldername * "/" * "$(f.filename).pdf")
-            end
-            rm(temp_dir, recursive=true)
+            rm("$(filename).tex")
```
[the remark this answers is not in the document]
Comment (Shushman) [reply src/TikzPictures.jl]: So the test has a `save(SVG)` followed by a `save(PDF)` and the files that remain are `testPic.svg`, `testPic.pdf`, and `testDoc.pdf`. Do you expect the `.tex` file to persist as well, considering `tikzDeleteIntermediate()` is set to `true` by default?

**Comment by Shushman** [comment]
Making a PR comment about a question I had - I believe that `Base.show`, at least when called from the `PGFPlots` notebook, looks for the `tmp_1000000x.png` file created by `PGFPlots` (which is sent to it as `tp.data`?).
I think the `.png` is always created in the current directory, which means that the `save` called inside `Base.show` should also be in the current directory. Since the `save(SVG)` is called with filename `"tikzpicture"` without any prefixed foldername, is that always going to be fine?

File: src/TikzPictures.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (tawheeler) [reply src/TikzPictures.jl]: Good. We don't want the `.tex` file to persist.

[the remark this answers is not in the document]
Comment (Shushman) [reply src/TikzPictures.jl]: Yeah. @mykelk spoke to me today and made a good point about having a temp directory, so I'll reintroduce the `tmpdir` stuff (but a bit simpler than what was there before)

**Comment by tawheeler** [comment]
IMO the package should not expose any files to the user unless the user asks for them. I'd change it so that `show` generates the file in a temp directory.

**Comment by mykelk** [comment]
Agreed with @tawheeler

**Comment by Shushman** [comment]
@mykelk has suggested I merge after the tests pass. I have changed `save(SVG)` to have independent behavior and also made `usePDF2SVG` a global flag

**Pull request merged**